lib/migration/config/v1/executor/container_vm.go

```diff
@@ -0,0 +1,218 @@
+
+	// The path in the executor at which this should be mounted
+	Path string `vic:"0.1" scope:"read-only" key:"dest"`
+
+	// Freeform mode string, which could translate directly to mount options
+	// We may want to turn this into a more structured form eventually
+	Mode string `vic:"0.1" scope:"read-only" key:"mode"`
+}
+
+// ContainerVM holds that data tightly associated with a containerVM, but that should not
+// be visible to the guest. This is the external complement to ExecutorConfig.
+type ContainerVM struct {
+	Common
+
+	// The version of the bootstrap image that this container was booted from.
+	Version string
+
+	// Name aliases for this specific container, Maps alias to unambiguous name
+	// This uses unambiguous name rather than reified network endpoint to persist
+	// the intent rather than a point-in-time manifesting of that intent.
+	Aliases map[string]string
+
+	// The location of the interaction service that the tether should connect to. Examples:
+	// * tcp://x.x.x.x:2377
+	// * vmci://moid - should this be an moid or a VMCI CID? Does one insulate us from reboots?
+	Interaction url.URL
+
+	// Key is the host key used during communicate back with the Interaction endpoint if any
+	// Used if the vSocket agent is responsible for authenticating the connection
+	AgentKey []byte
+}
+
+// ExecutorConfig holds the data tightly associated with an Executor. This is distinct from Sessions
+// in that there is no process inherently associated - this is closer to a ThreadPool than a Thread and
+// is the owner of the shared filesystem environment. This is the guest visible complement to ContainerVM.
+type ExecutorConfig struct {
+	Common `vic:"0.1" scope:"read-only" key:"common"`
+
+	// CreateTime stamp
+	CreateTime int64 `vic:"0.1" scope:"read-write" key:"createtime"`
+
+	// Diagnostics holds basic diagnostics data
+	Diagnostics Diagnostics `vic:"0.1" scope:"read-only" key:"diagnostics"`
+
+	// Sessions is the set of sessions currently hosted by this executor
+	// These are keyed by session ID
+	Sessions map[string]*SessionConfig `vic:"0.1" scope:"read-only" key:"sessions"`
+
+	// Maps the mount name to the detail mount specification
+	Mounts map[string]MountSpec `vic:"0.1" scope:"read-only" key:"mounts"`
+
+	// This describes an executors presence on a network, and contains sufficient
+	// information to configure the interface in the guest.
+	Networks map[string]*NetworkEndpoint `vic:"0.1" scope:"read-only" key:"networks"`
+
+	// Key is the host key used during communicate back with the Interaction endpoint if any
+	// Used if the in-guest tether is responsible for authenticating the connection
+	Key []byte `vic:"0.1" scope:"read-only" key:"key"`
+
+	// Layer id that is backing this container VM
+	LayerID string `vic:"0.1" scope:"read-only" key:"layerid"`
+
+	// Blob metadata for the caller
+	Annotations map[string]string `vic:"0.1" scope:"hidden" key:"annotations"`
+
+	// Repository requested by user
+	// TODO: a bit docker specific
```

Comment:
This TODO is unclear. It is also good practice to specify todo the following way:
TODO(emlin): ...commment...